rpc/regen.sh

```diff
@@ -7,7 +7,7 @@ build-protoc-gen-go() {
 }
 
 generate() {
-    protoc -I. api.proto --go_out=plugins=grpc:walletrpc
+    protoc -I. api.proto --go_out=plugins=grpc:walletrpc --go_opt=paths=source_relative
```

Comment:
When running this from within docker, it throws:
```
Sending build context to Docker daemon   7.81MB
Step 1/5 : FROM golang:1.12-stretch
 ---> 213fe73a3852
Step 2/5 : RUN apt-get update && apt-get install -y protobuf-compiler
 ---> Using cache
 ---> 4563e99fe378
Step 3/5 : RUN go get -u github.com/golang/protobuf/protoc-gen-go
 ---> Using cache
 ---> efcb6fb12aab
Step 4/5 : WORKDIR /build
 ---> Using cache
 ---> c89dd21bfa01
Step 5/5 : CMD "./regen.sh"
 ---> Using cache
 ---> 3187f85e9267
Successfully built 3187f85e9267
Successfully tagged protobuf-builder:latest
go: finding github.com/golang/protobuf/protoc-gen-go latest
go: finding github.com/golang/protobuf v1.3.5
go: downloading github.com/golang/protobuf v1.3.5
go: extracting github.com/golang/protobuf v1.3.5
Unknown flag: --go_opt
```

Without docker it does run correctly, however, the resulting `api.pb.go` fails to compile:
```
# decred.org/dcrwallet/rpc/walletrpc
rpc/walletrpc/api.pb.go:10876:11: undefined: grpc.SupportPackageIsVersion6
rpc/walletrpc/api.pb.go:10886:5: undefined: grpc.ClientConnInterface
```

Reply:
On an unrelated note, the shebang line is incorrect - it should be `#!/bin/bash` - perhaps this could be fixed while you're at it :)

Reply:
`bash` is not the only shell there is :)

To fix generation from within docker, see the fixup commit below (I'll squash it once you've reviewed it). 

I had to update the docker image to one with go1.14 and changed the script to mount the whole wallet source dir so that building `protoc-gen-go` would get the correct version from the main go.mod.

Reply:
> `bash` is not the only shell there is :)
> 
No of course, but it does functions better than `sh` ;)
```
./regen.sh
regen.sh: 3: regen.sh: Syntax error: Bad function name
```

> To fix generation from within docker, see the fixup commit below (I'll squash it once you've reviewed it).
> 
> I had to update the docker image to one with go1.14 and changed the script to mount the whole wallet source dir so that building `protoc-gen-go` would get the correct version from the main go.mod.

Awesome, that should sort things out. Thanks for looking into it!

Reply:
Looks all good on my end.

Reply:
```
$ /bin/bash
ksh: /bin/bash: not found
```

if you are going to require bash, at least do so portably (with /usr/bin/env).

Reply:
jrick: was that directed to the PR or just commenting on what peterzen wrote? 

The only reference to `/bin/bash` in the commit is on the Dockerfile which is based on debian buster so it's guaranteed that `/usr/bin` exists (nevertheless I can change it to using `env` if that's what you mean).

Reply:
that was to peterzen's comment.  If /bin/sh doesn't support all of the features (functions?) we need on all systems we develop from then we may want to change the shebang and shell, or rewrite it for portable POSIX shells.  

Keep in mind not all systems have bash installed (I have it installed on my openbsd systems but only to run the Go build scripts) and I strongly suspect that apple will eventually remove it from the macos distribution in the next year or two (zsh became the default shell in catalina, likely due to licensing.  macos bash was never updated past the last GPLv2 version).